client/reader/stream/post.jsx

```diff
@@ -196,6 +196,10 @@ const Post = React.createClass( {
 		// For Discover posts (but not site picks), open the original post in full post view
 		if ( this.state.originalPost ) {
 			postToOpen = this.state.originalPost;
+
+			if ( postToOpen ) {
+				postToOpen.referral = { blogId: DiscoverHelper.discoverBlogId, postId: this.props.ID }
```

Comment:
It looks like you are updating the state directly here. I think if you don't need `referral` to be in state then you should change the statement on line 198 to `postToOpen = { ...this.state.originalPost };`